CMakeLists.txt

```diff
@@ -192,8 +198,12 @@ else()
 endif()
 
 if(NOT WIN32)
-  core_optional_dep(OpenGl)
-  if(OPENGL_FOUND AND NOT APPLE)
+  if(NOT ROCKCHIP_FOUND)
+    core_optional_dep(OpenGl)
+  else()
+    core_optional_dep(XRandR ENABLE_X11)
+  endif()
+  if(OPENGL_FOUND AND NOT APPLE AND NOT ROCKCHIP_FOUND)
```

Comment:
not that constructive but you are right: it was a mess and now is even more a mess. i included and excluded the libs for rockchip so that i was sure it didnt break any possible other system (which i dont know and cant test) in line with the already existing spaghetti flow. so it just works without interfering with other any other possible display system builds.

my proposal would be to add a dedicated rockchip block just after "if(NOT WIN32)" that clearly and only includes all necessary libs for rockchip and doesnt enter the spaghetti flow further.

something like this:
```
if(NOT WIN32)
  if(ROCKCHIP_FOUND)
    core_require_dep(EGL)
    core_optional_dep(OpenGLES)
    core_optional_dep(X ENABLE_X11)
    core_optional_dep(XRandR ENABLE_X11)
    core_optional_dep(LibDRM ENABLE_X11)
  else()
    ... original code
```

would that be acceptable?